doc/doxygen/src/board-cpu-features.md

```diff
@@ -0,0 +1,99 @@
+
+FEATURES                                                             {#features}
+========
+
+`FEATURES_PROVIDED` are hardware (including toolchain) features (e.g.:`periph_hwrng`,
+`periph_uart`) that are available or characteristics (e.g:`arch_32bits`) of a board.
+
+Providing a `FEATURE`
+--------------------
+
+For a `FEATURE` to be provided by a `board` it must meet 2 criteria, and for
+`periph` and other `hw` it must follow the 3rd criteria.
+
+- Needs the "hardware" or BSP support (toolchain)
+    - e.g.: `stm32l152re` has an SPI peripheral
+            `riotboot` needs to be able to link and flash at an offset
+- Needs support in RIOT, an implementation of an api to interact with the `hw`
+    - e.g.: `cpu/stm32_common/periph/spi.c` is implemented for `stm32l1`
+            `riotboot` needs an implementation of `cpu_jump_to_image`
+- Wiring between the `cpu/soc` a `bus` and other `cpu`/`hw` components.
+    - e.g.: `nucleo-l152re/include/periph_conf.h` specified wiring between `PORT_Ax`
+             and `SPI1`
+
+What is a `FEATURE`
+--------------------
+
+Whenever a `FEATURE` is used there should be at some level a HW requirements,
+whether this is a `radio`, a `bus` ot a specific core architecture. 
+
+This is not a hard line, in some cases the line can be harder to establish than
+others. There are complicated cases like `netif` since a network interface could
+be fully implemented in software as a loop-back.
+
+It's also important to note that a `FEATURE` does not mean there is a `MODULE`
+with the same name. There could be many implementations for the same `FEATURE`.
+The fact that many `FEATURES` translate directly into a `MODULE` is only by
+convenience.
+
+e.g.
+
+    # all periph features correspond to a periph submodule
+    USEMODULE += $(filter periph_%,$(FEATURES_USED))
+
+A `FEATURE` is a tool to specify valid/invalid dependencies and configurations.
+
+
+CPU/CPU_MODEL                                                             {#cpu}
+======
+
+`CPU` and `CPU_MODEL` refer to the `SOC` (system on a chip) or `MCU` (microcontroller)
```

Comment:
SOC was already used earlier but not defined. I would move the definition when it's used the first time. Also no need to "highlight" it with backticks. Same for MCU. Maybe use italic ?